lib/internal/readline/eventsToAsyncIteratorFactory.js

```diff
@@ -0,0 +1,114 @@
+'use strict';
+const {
+  Promise,
+  SymbolAsyncIterator,
+} = primordials;
+
+const PAUSE_THRESHOLD = 1024;
+const RESUME_THRESHOLD = 1;
+const ITEM_EVENTS = ['data'];
+const CLOSE_EVENTS = ['close', 'end'];
+const ERROR_EVENTS = ['error'];
+
+let FixedQueue;
+let setImmediate;
+function waitNext(emitter, next, events) {
+  return new Promise((resolve, reject) => {
+    const resolveNext = () => {
+      events.forEach((event) => emitter.off(event, resolveNext));
+      setImmediate(() => {
+        try {
+          resolve(next());
+        } catch (promiseError) {
```

Comment:
Where does this error originates from?

Reply:
This error originates from line 96, from the next call